website/pages/en/cookbook/how-to-secure-api-keys-using-nextjs-server-components.mdx

```diff
@@ -0,0 +1,125 @@
+---
+title: How to Secure API Keys Using Next.js Server Components
+---
+
+## Overview
+
+We can use [NextJS server components](https://nextjs.org/docs/app/building-your-application/rendering/server-components) to properly secure our API key from exposure in the front-end of our dapp. To further increase our API key security, we can also [restrict our API key to certain subgraphs or domains in Subgraph Studio](/cookbook/upgrading-a-subgraph/#securing-your-api-key).
```

Comment:
@bermchain – Just noticed we sometimes write "frontend", sometimes "front-end". Does it matter?

Reply:
@benface I think we can standardize as frontend

Reply:
Thanks @bermchain, I'll standardize all the docs in another PR and will also check other codebases. Not sure how to quickly search in blog posts though, since they are in Notion and it looks like the database search only searches in properties, not page content... maybe not worth it for the blog?